{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "6126efd9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## Catriona Miller - 17/06/2022\n",
    "# This script can be made much more efficient if I use a better finding algorithm than a double for loop \n",
    "# (e.g. change the snps to a sorted integer list and use a smarter algorithm than checking every entry). \n",
    "# However, for now it is good enough as it doesn't take too long to run (~1-2 minutes) and I wanted something quick to code.\n",
    "\n",
    "import pandas as pd\n",
    "import os\n",
    "# These variables should be changed depending on what you're running. Nothing else needs to be updated\n",
    "\n",
    "# The reference GRN where you are searching for regulation (beta) (on figshare. See manuscript for details on adult and fetal GRN.)\n",
    "inputGRN = 'significant_eqtls_fetal_brain_cortex.txt'\n",
    "# The file contains snps and genes (output of m3d; same as supplementary tables S4 and S5)\n",
    "inputData = 'level4_sig_interactions.txt'\n",
    "# Where you want to output the updated file with the regulation column\n",
    "outputData = 'level4_sig_interactions_fetal_reg.txt'\n",
    "\n",
    "outputGene = 'gene_position_level4_fetal.txt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "59cc7038",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\catri\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3165: DtypeWarning: Columns (4,5,6,7,9,10,12,16,17,18,21,23,25) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    }
   ],
   "source": [
    "data = pd.read_csv(inputGRN,sep =\"\\t\")\n",
    "snps = pd.DataFrame(data)\n",
    "\n",
    "m3d_data = pd.read_csv(inputData,sep=\"\\t\")\n",
    "m3d = pd.DataFrame(m3d_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "eed3fa2c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snp</th>\n",
       "      <th>gencode_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>adj_pval</th>\n",
       "      <th>beta</th>\n",
       "      <th>beta_se</th>\n",
       "      <th>maf</th>\n",
       "      <th>interaction_type</th>\n",
       "      <th>hic_score</th>\n",
       "      <th>...</th>\n",
       "      <th>gene_start</th>\n",
       "      <th>gene_end</th>\n",
       "      <th>distance</th>\n",
       "      <th>cell_lines</th>\n",
       "      <th>cell_line_hic_scores</th>\n",
       "      <th>expression</th>\n",
       "      <th>max_expressed_tissue</th>\n",
       "      <th>max_expression</th>\n",
       "      <th>min_expressed_tissue</th>\n",
       "      <th>min_expression</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs60914341</td>\n",
       "      <td>ENSG00000121310.16</td>\n",
       "      <td>ECHDC2</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>0.035413</td>\n",
       "      <td>0.223574</td>\n",
       "      <td>0.049243</td>\n",
       "      <td>0.24424</td>\n",
       "      <td>Trans-intrachromosomal</td>\n",
       "      <td>1.67</td>\n",
       "      <td>...</td>\n",
       "      <td>52895910</td>\n",
       "      <td>52927212</td>\n",
       "      <td>1947265.0</td>\n",
       "      <td>Germinal_zone_neurons_Won2016, Cortical_plate_...</td>\n",
       "      <td>1.0, 0.67</td>\n",
       "      <td>3.94494</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>3.94494</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>3.94494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs61782124</td>\n",
       "      <td>ENSG00000121310.16</td>\n",
       "      <td>ECHDC2</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>0.048041</td>\n",
       "      <td>0.275777</td>\n",
       "      <td>0.061964</td>\n",
       "      <td>0.191244</td>\n",
       "      <td>Trans-intrachromosomal</td>\n",
       "      <td>1.67</td>\n",
       "      <td>...</td>\n",
       "      <td>52895910</td>\n",
       "      <td>52927212</td>\n",
       "      <td>1342346.0</td>\n",
       "      <td>Cortical_plate_neurons_Won2016, Germinal_zone_...</td>\n",
       "      <td>1.0, 0.67</td>\n",
       "      <td>3.94494</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>3.94494</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>3.94494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs35375774</td>\n",
       "      <td>ENSG00000121310.16</td>\n",
       "      <td>ECHDC2</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.308079</td>\n",
       "      <td>0.060448</td>\n",
       "      <td>0.195853</td>\n",
       "      <td>Trans-intrachromosomal</td>\n",
       "      <td>0.67</td>\n",
       "      <td>...</td>\n",
       "      <td>52895910</td>\n",
       "      <td>52927212</td>\n",
       "      <td>1374106.0</td>\n",
       "      <td>Germinal_zone_neurons_Won2016, Cortical_plate_...</td>\n",
       "      <td>0.33, 0.33</td>\n",
       "      <td>3.94494</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>3.94494</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>3.94494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs34886807</td>\n",
       "      <td>ENSG00000121310.16</td>\n",
       "      <td>ECHDC2</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.308079</td>\n",
       "      <td>0.060448</td>\n",
       "      <td>0.195853</td>\n",
       "      <td>Trans-intrachromosomal</td>\n",
       "      <td>0.67</td>\n",
       "      <td>...</td>\n",
       "      <td>52895910</td>\n",
       "      <td>52927212</td>\n",
       "      <td>1373573.0</td>\n",
       "      <td>Germinal_zone_neurons_Won2016, Cortical_plate_...</td>\n",
       "      <td>0.33, 0.33</td>\n",
       "      <td>3.94494</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>3.94494</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>3.94494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs61782122</td>\n",
       "      <td>ENSG00000121310.16</td>\n",
       "      <td>ECHDC2</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>0.008007</td>\n",
       "      <td>0.303015</td>\n",
       "      <td>0.061257</td>\n",
       "      <td>0.182028</td>\n",
       "      <td>Trans-intrachromosomal</td>\n",
       "      <td>0.67</td>\n",
       "      <td>...</td>\n",
       "      <td>52895910</td>\n",
       "      <td>52927212</td>\n",
       "      <td>1372543.0</td>\n",
       "      <td>Germinal_zone_neurons_Won2016, Cortical_plate_...</td>\n",
       "      <td>0.33, 0.33</td>\n",
       "      <td>3.94494</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>3.94494</td>\n",
       "      <td>Fetal_Brain_Cortex_Walker2019</td>\n",
       "      <td>3.94494</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          snp          gencode_id    gene                         tissue  \\\n",
       "0  rs60914341  ENSG00000121310.16  ECHDC2  Fetal_Brain_Cortex_Walker2019   \n",
       "1  rs61782124  ENSG00000121310.16  ECHDC2  Fetal_Brain_Cortex_Walker2019   \n",
       "2  rs35375774  ENSG00000121310.16  ECHDC2  Fetal_Brain_Cortex_Walker2019   \n",
       "3  rs34886807  ENSG00000121310.16  ECHDC2  Fetal_Brain_Cortex_Walker2019   \n",
       "4  rs61782122  ENSG00000121310.16  ECHDC2  Fetal_Brain_Cortex_Walker2019   \n",
       "\n",
       "   adj_pval      beta   beta_se       maf        interaction_type hic_score  \\\n",
       "0  0.035413  0.223574  0.049243   0.24424  Trans-intrachromosomal      1.67   \n",
       "1  0.048041  0.275777  0.061964  0.191244  Trans-intrachromosomal      1.67   \n",
       "2  0.004451  0.308079  0.060448  0.195853  Trans-intrachromosomal      0.67   \n",
       "3  0.004451  0.308079  0.060448  0.195853  Trans-intrachromosomal      0.67   \n",
       "4  0.008007  0.303015  0.061257  0.182028  Trans-intrachromosomal      0.67   \n",
       "\n",
       "   ... gene_start  gene_end   distance  \\\n",
       "0  ...   52895910  52927212  1947265.0   \n",
       "1  ...   52895910  52927212  1342346.0   \n",
       "2  ...   52895910  52927212  1374106.0   \n",
       "3  ...   52895910  52927212  1373573.0   \n",
       "4  ...   52895910  52927212  1372543.0   \n",
       "\n",
       "                                          cell_lines cell_line_hic_scores  \\\n",
       "0  Germinal_zone_neurons_Won2016, Cortical_plate_...            1.0, 0.67   \n",
       "1  Cortical_plate_neurons_Won2016, Germinal_zone_...            1.0, 0.67   \n",
       "2  Germinal_zone_neurons_Won2016, Cortical_plate_...           0.33, 0.33   \n",
       "3  Germinal_zone_neurons_Won2016, Cortical_plate_...           0.33, 0.33   \n",
       "4  Germinal_zone_neurons_Won2016, Cortical_plate_...           0.33, 0.33   \n",
       "\n",
       "  expression           max_expressed_tissue max_expression  \\\n",
       "0    3.94494  Fetal_Brain_Cortex_Walker2019        3.94494   \n",
       "1    3.94494  Fetal_Brain_Cortex_Walker2019        3.94494   \n",
       "2    3.94494  Fetal_Brain_Cortex_Walker2019        3.94494   \n",
       "3    3.94494  Fetal_Brain_Cortex_Walker2019        3.94494   \n",
       "4    3.94494  Fetal_Brain_Cortex_Walker2019        3.94494   \n",
       "\n",
       "            min_expressed_tissue min_expression  \n",
       "0  Fetal_Brain_Cortex_Walker2019        3.94494  \n",
       "1  Fetal_Brain_Cortex_Walker2019        3.94494  \n",
       "2  Fetal_Brain_Cortex_Walker2019        3.94494  \n",
       "3  Fetal_Brain_Cortex_Walker2019        3.94494  \n",
       "4  Fetal_Brain_Cortex_Walker2019        3.94494  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snps.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "c9ca9ef1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snp</th>\n",
       "      <th>trait</th>\n",
       "      <th>gene</th>\n",
       "      <th>eqtl_type</th>\n",
       "      <th>chromq</th>\n",
       "      <th>posq</th>\n",
       "      <th>rsidq</th>\n",
       "      <th>chromt</th>\n",
       "      <th>post</th>\n",
       "      <th>rsidt</th>\n",
       "      <th>corr</th>\n",
       "      <th>dprime</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs910049</td>\n",
       "      <td>Hepatitis C induced liver cirrhosis</td>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>spatial</td>\n",
       "      <td>6.0</td>\n",
       "      <td>32316388.0</td>\n",
       "      <td>rs3117123</td>\n",
       "      <td>6.0</td>\n",
       "      <td>32315726.0</td>\n",
       "      <td>rs910049</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs910049</td>\n",
       "      <td>Hepatitis C induced liver cirrhosis</td>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>spatial</td>\n",
       "      <td>6.0</td>\n",
       "      <td>32315506.0</td>\n",
       "      <td>rs3117127</td>\n",
       "      <td>6.0</td>\n",
       "      <td>32315726.0</td>\n",
       "      <td>rs910049</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs910049</td>\n",
       "      <td>Hepatitis C induced liver cirrhosis</td>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>spatial</td>\n",
       "      <td>6.0</td>\n",
       "      <td>32315100.0</td>\n",
       "      <td>rs3117129</td>\n",
       "      <td>6.0</td>\n",
       "      <td>32315726.0</td>\n",
       "      <td>rs910049</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs910049</td>\n",
       "      <td>Hepatitis C induced liver cirrhosis</td>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>spatial</td>\n",
       "      <td>6.0</td>\n",
       "      <td>32312081.0</td>\n",
       "      <td>rs6904320</td>\n",
       "      <td>6.0</td>\n",
       "      <td>32315726.0</td>\n",
       "      <td>rs910049</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs910049</td>\n",
       "      <td>Hepatitis C induced liver cirrhosis</td>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>spatial</td>\n",
       "      <td>6.0</td>\n",
       "      <td>32312352.0</td>\n",
       "      <td>rs7383144</td>\n",
       "      <td>6.0</td>\n",
       "      <td>32315726.0</td>\n",
       "      <td>rs910049</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        snp                                trait      gene eqtl_type  chromq  \\\n",
       "0  rs910049  Hepatitis C induced liver cirrhosis  HLA-DRB1   spatial     6.0   \n",
       "1  rs910049  Hepatitis C induced liver cirrhosis  HLA-DRB1   spatial     6.0   \n",
       "2  rs910049  Hepatitis C induced liver cirrhosis  HLA-DRB1   spatial     6.0   \n",
       "3  rs910049  Hepatitis C induced liver cirrhosis  HLA-DRB1   spatial     6.0   \n",
       "4  rs910049  Hepatitis C induced liver cirrhosis  HLA-DRB1   spatial     6.0   \n",
       "\n",
       "         posq      rsidq  chromt        post     rsidt  corr  dprime  \n",
       "0  32316388.0  rs3117123     6.0  32315726.0  rs910049   1.0     1.0  \n",
       "1  32315506.0  rs3117127     6.0  32315726.0  rs910049   1.0     1.0  \n",
       "2  32315100.0  rs3117129     6.0  32315726.0  rs910049   1.0     1.0  \n",
       "3  32312081.0  rs6904320     6.0  32315726.0  rs910049   1.0     1.0  \n",
       "4  32312352.0  rs7383144     6.0  32315726.0  rs910049   1.0     1.0  "
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "m3d.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "c975ecea",
   "metadata": {},
   "outputs": [],
   "source": [
    "regulation = []\n",
    "genestart = []\n",
    "geneend = []\n",
    "genechr = []\n",
    "typ = []\n",
    "i=0\n",
    "for snp in m3d['rsidq']:\n",
    "    j=0\n",
    "    for reference in snps['snp']:\n",
    "        if snp == reference:\n",
    "            if m3d['gene'][i] == snps['gene'][j]:\n",
    "                reg = snps['beta'][j]\n",
    "                start = snps['gene_start'][j]\n",
    "                end = snps['gene_end'][j]\n",
    "                chrom = snps['gene_chr'][j]\n",
    "                inter_typ = snps['interaction_type'][j]\n",
    "                regulation.append(reg)\n",
    "                genestart.append(start)\n",
    "                geneend.append(end)\n",
    "                genechr.append(chrom)\n",
    "                typ.append(inter_typ)\n",
    "        j=j+1\n",
    "    i=i+1\n",
    "\n",
    "m3d['regulation'] = regulation\n",
    "m3d['genestart'] = genestart\n",
    "m3d['geneend'] = geneend\n",
    "m3d['genechr'] = genechr\n",
    "m3d['interaction_type'] = typ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "51bbb4c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "m3d.to_csv(outputData,sep=\"\\t\",index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4df2a5d9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "ae66a525",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>genechr</th>\n",
       "      <th>genestart</th>\n",
       "      <th>geneend</th>\n",
       "      <th>regulation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>32578769</td>\n",
       "      <td>32589848</td>\n",
       "      <td>-0.449394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>32578769</td>\n",
       "      <td>32589848</td>\n",
       "      <td>-0.449394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>32578769</td>\n",
       "      <td>32589848</td>\n",
       "      <td>-0.449394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>32578769</td>\n",
       "      <td>32589848</td>\n",
       "      <td>-0.450296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>32578769</td>\n",
       "      <td>32589848</td>\n",
       "      <td>-0.449394</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene genechr genestart   geneend regulation\n",
       "0  HLA-DRB1    chr6  32578769  32589848  -0.449394\n",
       "1  HLA-DRB1    chr6  32578769  32589848  -0.449394\n",
       "2  HLA-DRB1    chr6  32578769  32589848  -0.449394\n",
       "3  HLA-DRB1    chr6  32578769  32589848  -0.450296\n",
       "4  HLA-DRB1    chr6  32578769  32589848  -0.449394"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Rearranging the dataframe so that it is suitable input for ChromoMap\n",
    "genepos = m3d.drop(columns=['trait','eqtl_type','snp','interaction_type'])\n",
    "order = ['gene','genechr','genestart','geneend','regulation']\n",
    "genepos = genepos[order]\n",
    "genepos.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "0a3783c0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>genechr</th>\n",
       "      <th>genestart</th>\n",
       "      <th>geneend</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>32578769</td>\n",
       "      <td>32589848</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>TRAF1</td>\n",
       "      <td>chr9</td>\n",
       "      <td>120902393</td>\n",
       "      <td>120929173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>IQGAP1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>90388218</td>\n",
       "      <td>90502243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>DHX33</td>\n",
       "      <td>chr17</td>\n",
       "      <td>5440912</td>\n",
       "      <td>5469060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277</th>\n",
       "      <td>BAZ1A</td>\n",
       "      <td>chr14</td>\n",
       "      <td>34752731</td>\n",
       "      <td>34862322</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene genechr  genestart    geneend\n",
       "0    HLA-DRB1    chr6   32578769   32589848\n",
       "78      TRAF1    chr9  120902393  120929173\n",
       "103    IQGAP1   chr15   90388218   90502243\n",
       "191     DHX33   chr17    5440912    5469060\n",
       "277     BAZ1A   chr14   34752731   34862322"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#TURN ON OR OFF IF REQUIRED\n",
    "genepos = genepos.drop(columns=['regulation'])\n",
    "genepos = genepos.drop_duplicates()\n",
    "genepos.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "bf47a774",
   "metadata": {},
   "outputs": [],
   "source": [
    "genepos.to_csv(outputGene,sep=\"\\t\",index=None,header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d9dbe00f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
